### DISCUSSION TITLE
Support BRC20 on Bitcoin

### DISCUSSION THREAD

**Discussion by lunarcatbt** [trustwallet/wallet-core#3166] [Q&A]
I'm trying to start implementing brc20 support based on your great foundation and then I ran into the very first step by generating `Taproot address`

what I did are 

1. added a new derivation entity to Bitcoin in `register.json`
2. add a new class `TaprootAddress` which is similar to `SegwitAddress` under the same folder use C++

I have run `boostrap.sh` to generate a new enum from Derivation and `tools/generate-files` to get all necessary files on iOS / Android

However, I was stucked by using the Taproot Class on Swift coz it doesn't generated any Swift file for this, may I ask what steps do I missed?

BTW, could you share some plans if you guys are working on this too?

**Comment by lamafab** [comment]
Hello lunarcatbt, see https://github.com/trustwallet/wallet-core/pull/3168: just pull/rebase on `master` and try again.

**Reply by lunarcatbt** [reply]
Thank you for the advice, but it seems my issue persists.

I might actually need a guide about how to add files to existing blockchains to extend their capability, I have tried

1. added a new derivation entity to Bitcoin in `register.json`
2. add a new C++ file under `src/bitcoin`
3. run `tools/generate-files`
4. no sign of new files in Swift

I couldn't find any match case in Docs, it would be fantasy if you can help to share some guideline.

**Reply by lamafab** [reply]
Hey, can you link me to your branch so I can have a look myself?

**Reply by lunarcatbt** [reply]
Sure, please check my branch https://github.com/lunarcatbt/wallet-core/tree/feature/taproot-address

**Reply by lamafab** [reply]
Hey. In order for `./tools/generate-files` to generate bindings to other languages for you, you need to create the interfaces that use your implementation:

* Create functions that wrap around your implementation in `src/interface/`, however you see fit.
* Write function headers in `include/`.

See [`src/interface/TWSegwitAddress.cpp`](https://github.com/trustwallet/wallet-core/blob/master/src/interface/TWSegwitAddress.cpp) and [`include/TWSegwitAddress.h`](https://github.com/trustwallet/wallet-core/blob/master/include/TrustWalletCore/TWSegwitAddress.h) for inspiration.

**Reply by lamafab** [reply]
@lunarcatbt just FYI: https://github.com/trustwallet/wallet-core/issues/3178

**Reply by leonis1** [reply]
thanks

**Comment by lamafab** [comment]
BRC20 support was just merged: https://github.com/trustwallet/wallet-core/pull/3233
